modules/ROOT/pages/dataweave-extension-plugin.adoc

```diff
@@ -0,0 +1,341 @@
+
+The tests use functions of the modules `dw::test::Tests` and `dw::test::Asserts` provided by the artifact `org.mule.weave.wtf` of the
+//xref:dataweave-testingframework.adoc[DataWeave Testing Framework].
+
+To generate a unit test for a DataWeave module, follow these steps:
+
+. In Visual Studio Code, open your DataWeave project.
+. Navigate to the `src/main/dw` folder.
+. Select your module `.dwl` file.
+. In your module `.dwl` file, click the *Add Unit Test* CodeLens button located above your function.
++
+image::dataweave-extension-testmodule-1.png[MyModule.dwl file showing the function and the Add Unit Test button]
+
+//image::animations/moduleTest.gif[Test Module Steps]
+// ![](animations/moduleTest.gif)
+
+This behavior generates:
+
+* A new `.dwl` test file for the module under the `src/test/dw` folder. The test file defines a test case for the function that you call with specific arguments and make assertions over the output.
+* Either a new test suite for the function, consisting of your new test case, or a new test case to an existing test suite.
+// > Dev team please define/explain better test file vs test case vs test suite
+
+image::dataweave-extension-testmodule-2.png[MyModuleTest.dwl file in the src/test/dw folder]
+
+To run the test:
+
+. In Visual Studio Code, go to the *TESTING* tab.
+. Select the dropdown arrows of your module test file.
+. Click the *Run Test* arrow button.
+
+image::dataweave-extension-testmodule-3.png[MyModuleTest file in the TESTING TAB and Run Test button]
+
+
+== Create a DataWeave Mapping
+
+A DataWeave mapping is a `.dwl` file that enables you to map and connect to anything, regardless of data type or structure. As when using a DataWeave module, you can use a mapping to define functions, variables, types, and namespaces, but you can additionally specify a body section after the separator `---`.
+
+Unlike a DataWeave module file, a DataWeave mapping file is an executable unit that enables you to transform zero or more inputs into a single output. Using the DataWeave 2.0 extension, you can develop these files to create reusable assets that you can deploy and share with others.
+
+To create a new DataWeave mapping, follow these steps:
+
+. In Visual Studio Code, launch *Command Palette* by pressing `Ctrl + Shift + P` (Windows, Linux) or `⇧⌘P` (macOS).
+. In the search bar, type `dataweave`.
+. Select *DataWeave: Create New Mapping*.
+
+image::dataweave-extension-createmapping.png[Command Palette with DataWeave: Create new mapping selected]
+//image::animations/new_mapping.gif[Creating new mapping using DataWeave Extension Plugin]
+// ![](animations/new_mapping.gif)
+
+[[sample-data]]
+=== Define Sample Data for DataWeave Mappings
+
+Generate multiple sample data inputs for your mapping by clicking the *Define Sample Data* CodeLens button at the top of your mapping `.dwl` file. Sample data is a key part of DataWeave project development, because it is used to run your mappings and also for suggestions.
```

Comment:
Do people know what CodeLens buttons are? I had to look it up. But maybe that's just me. (ha,ha)

Reply:
I also had to look it up in VS Code documentation. I think we can remove the wording "CodeLens" @dukesphere 
The screenshots highlight the buttons.

Reply:
It is a concept introduced by VSCode for links inside the code editor that trigger different actions.
People not coming from vscode environments might have trouble understanding what a CodeLens is. Should we add a short description somewhere?